core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java

```diff
@@ -466,87 +466,126 @@ default void createResponseFromRestMethod(final AnnotationScannerContext context
-                    content.addMediaType(producesType, mt);
-                }
+    default boolean hasResponseContent(Operation operation, String responseCode) {
+        APIResponse response = ModelUtil.responses(operation).getAPIResponse(responseCode);
+        return response != null && response.getContent() != null;
+    }
 
-                response.setContent(content);
+    default Content createEmptyResponseContent(AnnotationScannerContext context, MethodInfo method) {
+        Content content = OASFactory.createContent();
+        String[] produces = getProduces(context, method);
+
+        for (String mediaType : produces) {
+            MediaType model = OASFactory.createMediaType();
+            if ("application/json".equals(mediaType)) {
+                /*
+                 * Set an empty schema for JSON only to satisfy TCK requirement.
+                 * Remove this depending on the resolution of
+                 * https://github.com/microprofile/microprofile-open-api/issues/701
+                 */
+                model.setSchema(OASFactory.createSchema());
             }
+            content.addMediaType(mediaType, model);
         }
 
-        if (response != null) {
-            APIResponses responses = ModelUtil.responses(operation);
+        return content;
+    }
 
-            if (responses.hasAPIResponse(code)) {
-                APIResponse responseFromAnnotations = responses.getAPIResponse(code);
-                // Overlay the information from the annotations (2nd arg) onto the generated details (1st)
-                response = MergeUtil.mergeObjects(response, responseFromAnnotations);
-            }
+    default Content createResponseContentFromRestMethod(final AnnotationScannerContext context,
+            final MethodInfo method,
+            Type returnType) {
+        final Schema schema;
 
-            responses.addAPIResponse(code, response);
+        if (isMultipartOutput(returnType)) {
+            schema = OASFactory.createSchema().addType(Schema.SchemaType.OBJECT);
+        } else if (hasKotlinContinuation(method)) {
+            schema = kotlinContinuationToSchema(context, method);
+        } else {
+            schema = SchemaFactory.typeToSchema(context, returnType, null);
+        }
+
+        Content content = OASFactory.createContent();
+        String[] produces = getProduces(context, method);
+
+        BeanValidationScanner.applyConstraints(context, returnType, schema);
+        maybeUpdateOptionalResponseSchema(returnType, schema);
+
+        for (String producesType : produces) {
+            MediaType mt = OASFactory.createMediaType();
+            mt.setSchema(schema);
+            content.addMediaType(producesType, mt);
+        }
+
+        return content;
+    }
+
+    default void maybeUpdateOptionalResponseSchema(Type returnType, Schema schema) {
+        if (schema != null && SchemaSupport.getNullable(schema) == null
+                && TypeUtil.isOptional(returnType)) {
```

Comment:
It's not obvious what this method does from its name alone.

Would it be clearer if the `if` block here was removed, and the method was named `makeSchemaOptional`? The caller would then need to do the check in the `if` block before calling it.

Otherwise I think we should add doc here to say what this method does.